kano/network.py

```diff
@@ -535,6 +546,75 @@ def reload_kernel_module(device_vendor='148f', device_product='5370', module='rt
     return reloaded
 
 
+def connection_result(iface, wpa_file, connection_timeout, verbose=False):
+    '''
+    This function connects to the WPA supplicant event flow to detect the result of a connection attempt.
+    It returns a return code with an indication of the connection result. See the RC_* constants.
+
+    This code uses heuristics based on the supplicant source code: https://w1.fi/wpa_supplicant/
+    '''
+    rc=None
+    connected=False
+    scans = 0
+
+    def is_internet_up(monitor_file=INTERNET_UP_FILE):
+        return os.path.isfile(monitor_file)
+
+    run_cmd("wpa_supplicant -D nl80211,wext -t -d -c%s -i%s -f /var/log/kano_wpa.log -B" % (wpa_file, iface))
+
+    cli=subprocess.Popen(['wpa_cli'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE)
+    time.sleep(1)
+    rc=cli.poll()
+    while rc == None:
+        output = cli.stdout.readline().replace('\n', '')
+        print '>>> {} <<<'.format(output)
+
+        if output.find('CTRL-EVENT-CONNECTED') != -1:
+            print '[[[ Associated ]]]'
+            cli.stdin.write('quit\n')
+            cli.stdin.flush()
+            rc=RC_CONNECTED
+            break
+        elif output.find('reason=WRONG_KEY') != -1:
+            print '[[[ Wrong key ]]]'
+            cli.stdin.write('quit\n')
+            cli.stdin.flush()
+            rc=RC_BAD_PASSWORD
+            break
+        elif output.find('reason=CONN_FAILED') != -1:
+            print '[[[ AP not in range ]]]'
+            cli.stdin.write('quit\n')
+            cli.stdin.flush()
+            rc=RC_AP_NOT_IN_RANGE
+            break
+
+        if output.find('WPS-AP-AVAILABLE') != -1:
+            scans += 1
+            if scans == 5:
+                print '[[[ timeout due to too much scanning, router not in range ]]]'
+                cli.stdin.write('quit\n')
+                cli.stdin.flush()
+                reason_code = RC_AP_NOT_IN_RANGE
+                break
+
+        rc=cli.poll()
+
+    # If we are associated, wait for DHCP lease to become available
+    if rc==0:
```

Comment:
Should this if not be `if rc == RC_CONNECTED`. Exactly the same here as `RC_CONNECTED = 0` but still...

Reply:
absolutely. fixed